### PR TITLE
Armadillo: bump

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

File: sci-libs/armadillo/armadillo-8.300.4.recipe [hunk 1]
```diff
@@ -11,7 +11,7 @@ COPYRIGHT="2008-2017 Conrad Sanderson
 LICENSE="Apache v2"
 REVISION="2"
```
Comment (fbrosson) [inline sci-libs/armadillo/armadillo-8.300.4.recipe]:
```diff
-REVISION="2"
+REVISION="1"
```

File: sci-libs/armadillo/armadillo-8.400.0.recipe [hunk 2]
```diff
@@ -9,9 +9,9 @@ COPYRIGHT="2008-2017 Conrad Sanderson
...
 	2017 Data61, CSIRO"
 LICENSE="Apache v2"
-REVISION="2"
+REVISION="1"
 SOURCE_URI="https://downloads.sourceforge.net/arma/armadillo-$portVersion.tar.xz"
-CHECKSUM_SHA256="b41886f4b4ec8eab2b97088834974e5479f2e2075e01859832d6516db77e655f"
+CHECKSUM_SHA256="5cb6bc2f457a9d6a0758cfb15c418d48289909daccd79d0e428452029285dd9b"
 
 ARCHITECTURES="!x86_gcc2 x86 x86_64"
 SECONDARY_ARCHITECTURES="x86"
```
Comment (alaviss) [inline sci-libs/armadillo/armadillo-8.400.0.recipe]: A bit of `libVersionCompat` magic.

```diff
+
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}
 
 PROVIDES="
 	armadillo$secondaryArchSuffix = $portVersion
-	lib:libarmadillo$secondaryArchSuffix = $portVersion compat >= 7.800.2
+	lib:libarmadillo$secondaryArchSuffix = $libVersionCompat
```

```diff
 
 PROVIDES_devel="
 	armadillo${secondaryArchSuffix}_devel = $portVersion
-	devel:libarmadillo$secondaryArchSuffix = $portVersion compat >= 7.800.2
+	devel:libarmadillo$secondaryArchSuffix = $libVersionCompat
 	"
 REQUIRES_devel="
 	armadillo$secondaryArchSuffix == $portVersion base

File: sci-libs/armadillo/armadillo-8.400.0.recipe [hunk 3]
```diff
@@ -9,9 +9,9 @@ COPYRIGHT="2008-2017 Conrad Sanderson
 	2017 Arroyo Consortium
 	2017 Data61, CSIRO"
```
Comment (alaviss) [inline sci-libs/armadillo/armadillo-8.400.0.recipe]:
```diff
-COPYRIGHT="2008-2017 Conrad Sanderson
+COPYRIGHT="2008-2018 Conrad Sanderson
 	2008-2016 National ICT Australia (NICTA)
-	2017 Arroyo Consortium
-	2017 Data61, CSIRO"
+	2017-2018 Arroyo Consortium
+	2017-2018 Data61, CSIRO"
```

**Comment by extrowerk** [comment]
Tested by me

**Pull request merged**